packages/block-editor/src/components/block-list/insertion-point.js

```diff
@@ -200,6 +200,18 @@ function InsertionPointPopover( {
 		}
 	}
 
+	// Only show the inserter when there's a `nextElement` (a block after the
+	// insertion point). At the end of the block list the trailing appender
+	// should serve the purpose of inserting blocks.
+	const showInsertionPointInserter =
+		! isHidden && nextElement && ( isInserterShown || isInserterForced );
+
+	// Show the indicator if the insertion point inserter is visible, or if
+	// the `showInsertionPoint` state is `true`. The latter is generally true
+	// when hovering blocks for insertion in the block library.
+	const showInsertionPointIndicator =
+		! isHidden && ( showInsertionPointInserter || showInsertionPoint );
```

Comment:
`! isHidden` logic is duplicated, but I think this is one of those situations where avoiding that would make the code much uglier/harder to follow.

Reply:
`showInsertionPointInserter || ( ! isHidden && showInsertionPoint )`?

Reply:
Yeah, that's better 😄